homeassistant/components/light/mqtt.py

```diff
@@ -192,6 +192,10 @@ def turn_on(self, **kwargs):
         """Turn the device on."""
         should_update = False
 
+        if not self._state:
```

Comment:
I do not like this change as it will break other use cases:

For example, if the MQTT light is optimistic, our `self._state` variable is an assumption and the user is offered both turn on and turn off buttons in the UI. Now they will no longer work.

I think what you want instead is to only publish to a brightness topic and not publish to the turn on state topic ?